### PR TITLE
You can now feed bananas to monkey plants

### PR DESCRIPTION
**Pull request by tf-4**
## About The Pull Request

Using a banana on a monkey plant now feeds it to the plant, dropping the peel

I'm aware this is kind of a janky way to do it but I don't want to refactor potted plant spawn code

https://user-images.githubusercontent.com/8881105/169412324-4155f807-ac5b-4901-aec6-4d8222c9c485.mp4

## How This Contributes To The Skyrat Roleplay Experience

Very immersive and HRP

## Changelog
:cl:
add: You can now feed bananas to monkey plants.
fix: Fixes some broken icons on unused plant types, good if they're ever used
/:cl:

### CODE REVIEW

File: code/game/objects/structures/flora.dm [hunk 1]
```diff
@@ -359,6 +362,27 @@
 		if(do_after(user,3 SECONDS,target=src))
 			to_chat(user,span_notice("You finish trimming [src]."))
 			change_visual()
+	// SKYRAT EDIT START - OOK CHEE
+	if(icon_state == "monkeyplant")
```
Comment (Zonespace27) [inline code/game/objects/structures/flora.dm:366]: i am aware why you have to do this but it's not any less disgusting

File: code/game/objects/structures/flora.dm [hunk 2]
```diff
@@ -359,6 +362,27 @@
...
+		var/obj/item/food/grown/banana/banana = I
+		if(!istype(I))
+			return
+		if(!COOLDOWN_FINISHED(src, banana_consumption_cooldown))
+			balloon_alert(user, "not hungry!")
+			return
+		var/obj/item/grown/bananapeel/peel = new banana.trash_type(src.loc)
```
Comment (Zonespace27) [inline code/game/objects/structures/flora.dm]: might want to do `get_turf()` instead of `.loc`

File: code/game/objects/structures/flora.dm [hunk 3]
```diff
@@ -359,6 +362,27 @@
 		if(do_after(user,3 SECONDS,target=src))
 			to_chat(user,span_notice("You finish trimming [src]."))
 			change_visual()
+	// SKYRAT EDIT START - OOK CHEE
+	if(icon_state == "monkeyplant")
+		var/obj/item/food/grown/banana/banana = I
+		if(!istype(I))
```
Comment (Zonespace27) [inline code/game/objects/structures/flora.dm]: you sure it's intended to check the type of `I`, not `banana`?

File: code/game/objects/structures/flora.dm [hunk 4]
```diff
@@ -359,6 +362,27 @@
...
+			return
+		if(!COOLDOWN_FINISHED(src, banana_consumption_cooldown))
+			balloon_alert(user, "not hungry!")
+			return
+		var/obj/item/grown/bananapeel/peel = new banana.trash_type(src.loc)
+		qdel(banana)
+		to_chat(user, span_notice("[src] chews up [banana], spitting out [peel]!"))
+		playsound(src, pick(list('sound/creatures/monkey/monkey_screech_1.ogg',
```
Comment (Zonespace27) [inline code/game/objects/structures/flora.dm:376]: make this a variable and a static list

**Review by Zonespace27 — CHANGES_REQUESTED** [review]
May want to be taken upstream

**Comment by jjpark-kb** [comment]
I think it would be nice if it puts the banana peel in your hands as well.

File: code/game/objects/structures/flora.dm (hunk 1 above)
Comment (tf-4) [reply code/game/objects/structures/flora.dm:366]: yeah I hate it too. Might take it upstream but they don't even have any of the sprites and ehh. I'll come back to that one.

File: code/game/objects/structures/flora.dm (hunk 3 above)
Comment (tf-4) [reply code/game/objects/structures/flora.dm]: oops

Comment (tf-4) [inline code/game/objects/structures/flora.dm]:
```suggestion
		if(!istype(banana))
```

File: code/game/objects/structures/flora.dm [hunk 5]
```diff
@@ -359,6 +362,27 @@
...
 			change_visual()
+	// SKYRAT EDIT START - OOK CHEE
+	if(icon_state == "monkeyplant")
+		var/obj/item/food/grown/banana/banana = I
+		if(!istype(banana))
+			return
+		if(!COOLDOWN_FINISHED(src, banana_consumption_cooldown))
+			balloon_alert(user, "not hungry!")
+			return
+		var/obj/item/grown/bananapeel/peel = new banana.trash_type(src.loc)
```
Comment (tf-4) [inline code/game/objects/structures/flora.dm]:
```suggestion
		var/obj/item/grown/bananapeel/peel = new banana.trash_type(get_turf(src))
```

**Comment by tf-4** [comment]
this is gonna be too much shitcode I'm gonna do this upstream

**Pull request closed without merging**